### PR TITLE
Kotlin 2.0.0

### PR DESCRIPTION
**Pull request by rickclephas**

### CODE REVIEW

File: kmp-nativecoroutines-compiler/build.gradle.kts [hunk 1]
```diff
@@ -34,13 +34,15 @@ java {
...
-        freeCompilerArgs = listOf("-Xjvm-default=all")
+    compilerOptions {
+        freeCompilerArgs.add("-Xjvm-default=all")
     }
 }
 
 tasks.test {
-    systemProperty("kotlin.internal.native.test.nativeHome", NativeCompilerDownloader(project).compilerDirectory.absolutePath)
+    val compilerDownloader = NativeCompilerDownloader(project)
+    compilerDownloader.downloadIfNeeded()
```
Comment (rickclephas) [inline kmp-nativecoroutines-compiler/build.gradle.kts:44]: This isn't ideal need to find a way to depend on the new download logic

Comment (rickclephas) [reply kmp-nativecoroutines-compiler/build.gradle.kts:44]: https://youtrack.jetbrains.com/issue/KT-66929/Allow-Gradle-tasks-to-depend-on-native-compiler

**Comment by vanniktech** [comment]
New release is also coming? :shipit:

**Comment by rickclephas** [comment]
Yes, somewhere today 😇

**Pull request merged**